lib/features/auto-place/BpmnAutoPlace.js

```diff
@@ -1,18 +1,26 @@
-import { getNewShapePosition } from './BpmnAutoPlaceUtil';
+import { getNewShapePosition, getScrollOffset } from './BpmnAutoPlaceUtil';
 
 
 /**
  * BPMN auto-place behavior.
  *
  * @param {EventBus} eventBus
  */
-export default function AutoPlace(eventBus) {
+export default function AutoPlace(eventBus, canvas) {
```

Comment:
can you maybe explain, why this is a bpmn-js but no generic diagram-js concern?

Reply:
One could imagine this is a general create behavior. Elements created are always moved into sight. Or what exactly do you refer to here @pinussilvestrus?

Reply:
I see the benefits of moving this PR to diagram-js, since we also moved the AutoPlace behavior there with https://github.com/bpmn-io/dmn-js/issues/470

This would also add this feature to DMN modeling, which makes sense to me

Reply:
So we have two things here

a) is it a diagram-js concern? I'd vote for **yes**
b) is this an auto-place (only) behavior? Good question, I'd expect that this also happens on creating any kind of element, no matter when this happens (auto-place, creating from the palette, ...)

Reply:
AFAIK, auto-place is the only way we can currently place elements outside of the viewbox. When creating elements from the pallet, the drag operation automatically scrolls the canvas already

Reply:
@marstamm We could argue that is the case, even though we do allow anyone to _model via API_: https://cdn.statically.io/gh/nikku/2021-token-simulation/v0.0.2/presentation.html#22

Used by some of our side projects, e.g. https://github.com/bpmn-io/bpmn-js-cli and https://github.com/nikku/bpmn-js-cli-modeling-dsl.

We could argue that auto place is the natural higher level API that ships with such behavior. People and robots (cf. [WASDENN](https://bpmn.io/blog/posts/2021-wasdenn-ai-modeling-assistant.html)) are free to implement this this in a similar fashion.

Reply:
One idea: Can we make this "ensure visible" a thing on the `canvas` in diagram-js? Scroll to an element is a common use case people ask us about on our forums. It is also being used in [our search feature already](https://github.com/bpmn-io/diagram-js/blob/develop/lib/features/search-pad/SearchPad.js#L214). Moving it into a shared place and exposing it as an API would allow a broader user base to benefit.

Reply:
Great Idea, let's move it into the canvas :+1:

Reply:
@nikku , I'm not sure if we should replace the behavior in the search feature, as the current implementation is "center on screen" and not "scroll until visible". On the other hand, changing the behavior would closer resemble a text search, as no scrolling is happening when the element is already visible. What do you think?

Reply:
> On the other hand, changing the behavior would closer resemble a text search, as no scrolling is happening when the element is already visible. 

I think it makes a lot of sense to scroll a little bit less. The search result display is already very verbose. So :+1: from my perspective to scroll until visible only.

We may also consider to add a little bit more padding around the to be visible element. It does make sense to show it with context (i.e. what is behind). Would appreciate your input / if you experiment with it a little bit.